LibraryLoading.h

```diff
@@ -0,0 +1,10 @@
+#ifndef LIBRARY_LOADING_H
+#define LIBRARY_LOADING_H
+
+#include <string>
+
+void AddLibrarySearchPath(std::string path);
+// Returns true on success, false on failure
+bool SearchForAndLoadLibraryPermanently(std::string name);
```

Comment:
Maybe this could be an `llvm::Error`?

Reply:
Hmm well `llvm::sys::DynamicLibrary::LoadLibraryPermanently` returns true or false, so I wouldn't really know the exact type of error anyway (except maybe for something like a file not found error).